oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java

```diff
@@ -77,101 +77,280 @@ protected void createTestIndexNode() throws Exception {
         assertEquals("[]", log);
     }
 
-    private String runTest(Class<?> loggerClass, boolean nameProperty) throws CommitFailedException {
-        LogCustomizer customLogs = LogCustomizer
-                .forLogger(loggerClass)
-                .enable(Level.TRACE).create();
+    // verifying fulltext query syntax ability, put here for comparison with dynamic boost and dynamic boost lite
+    @Test
+    public void testQueryFullTextWithTitle() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'titleone')]", XPATH, Arrays.asList("/test/asset1"));
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'long')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+        // case insensitive
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'LONG')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+        // wildcard works
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'title*')]", XPATH,
+                Arrays.asList("/test/asset1", "/test/asset2", "/test/asset3"));
+        // space is treated as AND
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'long titleone')]", XPATH, Arrays.asList("/test/asset1"));
+        // OR works
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'long OR titleone')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+        // minus works
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'long -titleone')]", XPATH, Arrays.asList("/test/asset2"));
+    }
+
+    // Section 2. Query dynamicboost testing
+    @Test
+    public void testQueryDynamicBoostBasic() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertEquals(
+                "[dam:Asset] as [a] /* lucene:test-index(/oak:index/test-index) :fulltext:plant " + "((jcr:content/metadata/predictedTags/plant:1 jcr:content/metadata/predictedTags/plant:1)^1.0E-4) ft:(\"plant\")\n" + "  where contains([a].[*], 'plant') */",
+                explain("//element(*, dam:Asset)[jcr:contains(., 'plant')]", XPATH));
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'plant')]", XPATH,
+                Arrays.asList("/test/asset1", "/test/asset2", "/test/asset3"));
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'flower')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+    }
+
+    // dynamic boost: query should be case insensitive
+    @Test
+    public void testQueryDynamicBoostCaseInsensitive() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'FLOWER')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+    }
+
+    @Ignore //todo: bug? wildcard doesn't work at all for dynamic boost term
```

Comment:
this cannot work with standard dynamic boost since the terms are the field names. I would remove this test

Reply:
make sense, removed it